openedx/features/wikimedia_features/admin_dashboard/static/admin_dashboard/js/course_reports.js

```diff
@@ -25,29 +25,60 @@
             dataType: 'json',
             url: endpoint,
             success: function(data) {
-                $('#report-downloads-table').empty();
-                if (data.downloads.length){
+                if (data.downloads.length && $('#report-downloads-table').find('a').length == 0){
                     for (let i = 0; i < data.downloads.length; i++) {
                         $('#report-downloads-table').append(data.downloads[i]['link']);
                         $('#report-downloads-table').append('<br>');
                     }
                 }
+                else if (data.downloads.length > $('#report-downloads-table').find('a').length){
+                    let newReportsFetched = data.downloads.length - $('#report-downloads-table').find('a').length;
+                    for (let i = newReportsFetched; i > 0; i--) {
+                        $('#report-downloads-table').prepend('<br>');
+                        $('#report-downloads-table').prepend(data.downloads[i-1]['link']);
+                    }
+                }
             },
             error: function() {
                 console.log("There is an error in the list report downloads api")
             }
         });
     };
 
+    setInterval(function() {
+        ReportDownloads();
+    }, 20000);
+    
+
     $('select').change(function (e) {
         e.preventDefault();
-        let course_name = $(this).val();
-        url = '/courses/' + course_name + '/instructor/api/get_students_features' + '/csv';
-        endpoint = '/courses/' + course_name + '/instructor/api/list_report_downloads';
+        $('#report-request-response').empty();
+        $('#report-request-response-error').empty();
         $('#report-downloads-table').empty();
+        if ($(this).val())
+        {
+            if ($(this).val().length> 1) {
+                course_name = $(this).val().toString();
+                for (let i = 0; i < $('.single-course-report').length; i++){
```

Comment:
create separate method to update state and send "none" or "block" as parameters

Reply:
Resolved